pulsar/src/main/java/io/micronaut/pulsar/annotation/PulsarReader.java

```diff
@@ -0,0 +1,122 @@
+/*
+ * Copyright 2017-2020 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.pulsar.annotation;
+
+import io.micronaut.pulsar.config.AbstractPulsarConfiguration;
+import org.apache.pulsar.client.api.RegexSubscriptionMode;
+
+import javax.validation.constraints.Pattern;
+import java.lang.annotation.Documented;
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.Target;
+
+import static java.lang.annotation.RetentionPolicy.RUNTIME;
+
+/**
+ * Create and inject Pulsar reader into field.
+ *
+ * @author Haris Secic
+ * @since 1.0
+ */
+@Documented
+@Retention(RUNTIME)
+@Target({ElementType.FIELD})
+public @interface PulsarReader {
```

Comment:
Sounds good. Should I introduce also `String topic()` then? I used `topics` in hope to take less code space and let users use single element array. However this can be changed to make more sense.

Reply:
As mentioned above, no need to do that in firt PR.